lib/src/map/options/enabled_gestures.dart

```diff
@@ -0,0 +1,287 @@
+    bool? twoFingerZoom,
+    bool? twoFingerMove,
+    bool? doubleTapZoomIn,
+    bool? doubleTapDragZoom,
+    bool? scrollWheelZoom,
+    bool? twoFingerRotate,
+    bool? keyTriggerDragRotate,
+    bool? trackpadZoom,
+  }) =>
+      EnabledGestures(
+        drag: drag ?? this.drag,
+        flingAnimation: flingAnimation ?? this.flingAnimation,
+        twoFingerZoom: twoFingerZoom ?? this.twoFingerZoom,
+        twoFingerMove: twoFingerMove ?? this.twoFingerMove,
+        doubleTapZoomIn: doubleTapZoomIn ?? this.doubleTapZoomIn,
+        doubleTapDragZoom: doubleTapDragZoom ?? this.doubleTapDragZoom,
+        scrollWheelZoom: scrollWheelZoom ?? this.scrollWheelZoom,
+        twoFingerRotate: twoFingerRotate ?? this.twoFingerRotate,
+        keyTriggerDragRotate: keyTriggerDragRotate ?? this.keyTriggerDragRotate,
+        trackpadZoom: trackpadZoom ?? this.trackpadZoom,
+      );
+
+  const EnabledGestures.byGroup({
+    required bool move,
+    required bool zoom,
+    required bool rotate,
+  }) : this(
+          drag: move,
+          twoFingerMove: move,
+          flingAnimation: move,
+          doubleTapDragZoom: zoom,
+          doubleTapZoomIn: zoom,
+          scrollWheelZoom: zoom,
+          twoFingerZoom: zoom,
+          twoFingerRotate: rotate,
+          keyTriggerDragRotate: rotate,
+          trackpadZoom: zoom,
+        );
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is EnabledGestures &&
+          runtimeType == other.runtimeType &&
+          drag == other.drag &&
+          flingAnimation == other.flingAnimation &&
+          twoFingerMove == other.twoFingerMove &&
+          twoFingerZoom == other.twoFingerZoom &&
+          doubleTapZoomIn == other.doubleTapZoomIn &&
+          doubleTapDragZoom == other.doubleTapDragZoom &&
+          scrollWheelZoom == other.scrollWheelZoom &&
+          twoFingerRotate == other.twoFingerRotate &&
+          keyTriggerDragRotate == other.keyTriggerDragRotate;
+
+  @override
+  int get hashCode => Object.hash(
+        drag,
+        flingAnimation,
+        twoFingerMove,
+        twoFingerZoom,
+        doubleTapZoomIn,
+        doubleTapDragZoom,
+        scrollWheelZoom,
+        twoFingerRotate,
+        keyTriggerDragRotate,
+      );
+}
+
+/// Use [InteractiveFlag] to disable / enable certain events Use
+/// [InteractiveFlag.all] to enable all events, use [InteractiveFlag.none] to
+/// disable all events
+///
+/// If you want mix interactions for example drag and rotate interactions then
+/// you have two options:
+///   a. Add your own flags:
+///      [InteractiveFlag.drag] | [InteractiveFlag.twoFingerRotate]
+///   b. Remove unnecessary flags from all:
+///     [InteractiveFlag.all] &
+///       ~[InteractiveFlag.flingAnimation] &
+///       ~[InteractiveFlag.twoFingerMove] &
+///       ~[InteractiveFlag.twoFingerZoom] &
+///       ~[InteractiveFlag.doubleTapZoomIn]
+abstract class InteractiveFlag {
```

Comment:
As in https://github.com/fleaflet/flutter_map/pull/1733#discussion_r1428964382, we should probably remove this. I'm not sure if there's a convienient method of deprecation here, but this redundancy just complicates things IMO. WDYT?

Reply:
I don't care either way but @mootw had a strong preference for this. I let the discussion up to both of you.

Reply:
I believe Moo may now be busy for a while with other matters. Can you link to his discussion?

Reply:
Ok, let's leave it in for now.